grails-app/services/org/pih/warehouse/data/LocationDataService.groovy

```diff
@@ -33,32 +31,44 @@ class LocationDataService {
 
         command.data.eachWithIndex { params, index ->
 
-            Location location = createOrUpdateLocation(params)
-
+            Location newLocation = Location.findByName(params.name)
+            Location location = new Location()
```

Comment:
Is this just here to carry the errors?

Reply:
I think we can just use the command object for that unless you are rejecting specific fields.

Reply:
You are right, it was useful in my previous approach but finally it is not necessary.